File: appUtil/src/main/java/com/theroyalsoft/mydoc/apputil/baseUI/RoundShape.java
package com.theroyalsoft.mydoc.apputil.baseUI;

import android.graphics.Color;
import android.graphics.DashPathEffect;
import android.graphics.Paint;
import android.graphics.Path;

public class RoundShape {
    private Paint paint;
    private Path path;

    public RoundShape() {
        paint = new Paint();
        paint.setStrokeWidth(5);
        paint.setColor(Color.parseColor("#0098C9"));
        paint.setStyle(Paint.Style.STROKE);
        paint.setPathEffect(new DashPathEffect(new float[] {5f,5f}, 20f));


        path = new Path();
    }

    public void setCircle(float x, float y, float radius, Path.Direction dir) {
        path.reset();
        path.addCircle(x, y, radius, dir);
    }

    public Path getPath() {
        return path;
    }

    public Paint getPaint() {
        return paint;
    }

}

File: app/src/main/java/com/theroyalsoft/telefarmer/utils/RoundShape.kt
package com.theroyalsoft.telefarmer.utils

import android.graphics.Color
import android.graphics.DashPathEffect
import android.graphics.Paint
import android.graphics.Path

/**
 * Created by Pritom Dutta on 31/7/23.
 */
class RoundShape {
    private val paint: Paint
    private val path: Path

    init {
        paint = Paint()
        paint.strokeWidth = 5f
        paint.color = Color.parseColor("#0098C9")
        paint.style = Paint.Style.STROKE
        paint.pathEffect = DashPathEffect(floatArrayOf(5f, 5f), 20f)
        path = Path()
    }

    fun setCircle(x: Float, y: Float, radius: Float, dir: Path.Direction?) {
        path.reset()
        path.addCircle(x, y, radius, dir!!)
    }

    fun getPath(): Path {
        return path
    }

    fun getPaint(): Paint {
        return paint
    }
}
